eslint.config.js

```diff
@@ -0,0 +1,96 @@
+import eslint from '@eslint/js';
+import singlestoreReactHooksDisableImport from '@singlestore/eslint-plugin-react-hooks-disable-import';
+import eslintConfigPrettier from 'eslint-config-prettier';
+import jsxA11y from 'eslint-plugin-jsx-a11y';
+import react from 'eslint-plugin-react';
+import hooksPlugin from 'eslint-plugin-react-hooks';
+import reactRefresh from 'eslint-plugin-react-refresh';
+import sonarjs from 'eslint-plugin-sonarjs';
+import globals from 'globals';
+import tseslint from 'typescript-eslint';
+const __dirname = new URL('.', import.meta.url).pathname;
+
+export default [
+  {
+    files: ['**/*.ts', '**/*.tsx'],
+  },
+  {
+    ignores: ['**/dist', '**/eslint.config.js', 'src/api/schema.d.ts', '**/coverage', '**/postcss.config.js'],
+  },
+
+  eslint.configs.recommended,
+  jsxA11y.flatConfigs.recommended,
+  react.configs.flat.recommended,
+  react.configs.flat['jsx-runtime'],
+  sonarjs.configs.recommended,
+  ...tseslint.configs.recommended,
+  ...tseslint.configs.stylistic,
+  eslintConfigPrettier, // must be last, override other configs
+  {
+    plugins: {
+      '@singlestore/react-hooks-disable-import': singlestoreReactHooksDisableImport,
+      react: react,
+      'react-hooks': hooksPlugin,
+      'react-refresh': reactRefresh,
+    },
+  },
+  {
+    languageOptions: {
+      globals: {
+        ...globals.browser,
+      },
+
+      ecmaVersion: 5,
+      sourceType: 'script',
+
+      parserOptions: {
+        ecmaVersion: 2020,
+        sourceType: 'module',
+        project: ['./tsconfig.json', './tsconfig.node.json'],
+        tsconfigRootDir: __dirname,
+      },
+    },
+
+    settings: {
+      react: {
+        version: 'detect',
+      },
+    },
+
+    rules: {
+      'react-refresh/only-export-components': [
+        'warn',
+        {
+          allowConstantExport: true,
+        },
+      ],
+
+      'no-useless-rename': 'error',
+      'no-console': 'warn',
+      'react/jsx-child-element-spacing': 'error',
+      '@typescript-eslint/consistent-indexed-object-style': 'warn',
+      '@typescript-eslint/no-misused-promises': 'warn',
+      '@typescript-eslint/no-floating-promises': 'warn',
+      '@typescript-eslint/no-unused-vars': [
+        'error',
+        {
+          args: 'all',
+          argsIgnorePattern: '^_',
+          caughtErrors: 'all',
+          caughtErrorsIgnorePattern: '^_',
+          destructuredArrayIgnorePattern: '^_',
+          varsIgnorePattern: '^_',
+          ignoreRestSiblings: true,
+        },
+      ],
+      '@singlestore/react-hooks-disable-import/react-hooks-disable-import': 'error',
+      'sonarjs/new-cap': 'off',
+      'sonarjs/no-ignored-exceptions': 'off',
+      'sonarjs/todo-tag': 'warn',
```

Comment:
I still left this on for the purpose for really nagging about TODOs in code.
You have to explicitly disable the rule per file/line --> then there is two todo in the file -> another in caps and other without it. 
At least you know find those TODOs to get them fixed! 😁